src/main/java/ca/ciccc/Assignment1.java

```diff
@@ -89,7 +114,19 @@ public static String bmi(int kgs, double meters) {
      */
     public static String speed(int meters, int hours, int minutes, int seconds) {
 
-        return "";
+
+        float timeSeconds = (hours*3600) + (minutes*60) + seconds;
+        float mps = meters / timeSeconds;
+        float kph = (meters / 1000.0f) / (timeSeconds / 3600.0f);
+        float mph = kph / 1.609f;
+
+        DecimalFormat df = new DecimalFormat("#.####");
+        df.setRoundingMode(RoundingMode.DOWN);
+
+
+
+        String result = String.format("Your speed in meters/second is %s\nYour speed in km/h is %s\nYour speed in miles/h is %s", df.format(mps), df.format(kph), df.format(mph));
+        return result;
```

Comment:
For your information,

```
String.format("Your speed in meters/second is %1$s\nYour speed in km/h is %2$s\nYour speed in miles/h is %3$s", df.format(mps), df.format(kph), df.format(mph));
```

is also possible and sometimes easier to see 👍

Reply:
👍